src/hivpy/population.py

```diff
@@ -51,15 +51,29 @@ def _create_population_data(self):
         self.data[col.NUM_PARTNERS] = 0
         self.sexual_behaviour.init_sex_behaviour_groups(self.data)
         self.sexual_behaviour.init_risk_factors(self.data)
-        self.sexual_behaviour.num_short_term_partners(self.data)
+        self.sexual_behaviour.num_short_term_partners(self)
+
+    def transform_group(self, param_list, func, use_size=True, **kwargs):
+        """Groups the data by a list of parameters and applies a function to each grouping"""
+        # HIV_STATUS is just a dummy column to allow us to use the transform method
+        def general_func(g):
+            args = list(g.name)
+            if (use_size):
+                args.append(g.size)
+            return func(*args)
+        if "sub_pop" in kwargs.keys():
+            df = self.data.loc[kwargs["sub_pop"]]
```

Comment:
Since this is a fixed keyword argument that we (potentially) expect, we could make it explicitly part of the signature with a default value, like `use_size`. I think that would make it cleaner for people looking at how to call the function too.

Reply:
I did think about this, but the problem that I have here is that for something like `use_size` there is a trivial default, and I wasn't sure how to define a default so that the default sub-population is the whole population, and then that would lead to an unnecessary call to `loc` anyway. But I suppose what we could do is have a default `sub_pop=None`? And then presumably can test for `None` and apply `loc` only if the `sub_pop` is defined.

Reply:
Yes, that's what I had in mind. `None` is a common default in many Python frameworks to indicate that extra processing is not needed.